src/Components/Post/Footer/UpvoteButton.js

```diff
@@ -0,0 +1,198 @@
+    ) {
+      if (this.props.platform === "steem") {
+        this.setState({
+          weight: isUpvotedSteem(this.props.activeVotes)
+        });
+      } else if (this.props.platform === "email") {
+        this.setState({
+          weight: isUpvotedFirebase(this.props.upvotes)
+        });
+      }
+    }
+  }
+  handleVoteBtn = async () => {
+    const { permlink, location } = this.props;
+    const login = store.getState().login;
+    if (login.status) {
+      await this.handleVoting();
+      if (location === "blog") {
+        store.dispatch(updatePost(permlink));
+        store.dispatch(updatePostByAuthor(permlink));
+      } else {
+        store.dispatch(updatePost(permlink));
+      }
+    } else {
+      alert("You have to login first");
+    }
+  };
+  handleVoting = async () => {
+    const { platform, permlink, author } = this.props;
+    const { weight } = this.state;
+    const login = store.getState().login;
+    if (login.status && login.platform === "steem") {
+      //If user logged in with steem account
+      if (weight === 0) {
+        if (platform === "email") {
+          //voting on non-blockchain posts
+          await upvotePost({ permlink: permlink, weight: 1 });
+          this.setState({
+            weight: 1
+          });
+        } else if (platform === "steem") {
+          //voting on steem blockchain posts
+          await steemVote(
+            login.username,
+            author,
+            permlink,
+            store.getState().votePower.power
+          );
+          await upvoteSteemPost({
+            permlink: permlink,
+            weight: store.getState().votePower.power
+          });
+          this.props.updateValue(store.getState().votePower.power);
+          this.updateVotingState(
+            {
+              permlink: author + "/" + permlink,
+              percent: store.getState().votePower.power
+            },
+            true
+          );
+          this.setState({
+            weight: store.getState().votePower.power
+          });
+        }
+      } else if (weight > 0) {
+        //unvoting post
+        if (platform === "email") {
+          //voting on non-blockchain posts
+          await upvotePost({ permlink: permlink, weight: 0 });
+          this.setState({
+            weight: 0
+          });
+        } else if (platform === "steem") {
+          //unvoting steem blockchain post
+          await steemVote(login.username, author, permlink, 0);
+          await upvoteSteemPost({
+            permlink: permlink,
+            weight: 0
+          });
+          this.props.updateValue(0);
+          this.updateVotingState(
+            {
+              permlink: author + "/" + permlink,
+              percent: 0
+            },
+            false
+          );
+          this.setState({
+            weight: 0
+          });
+        }
+      }
+    } else if (login.status && login.platform === "email") {
+      //If user logged in with steem account
```

Comment:
Should be "logged in with email", right?

Reply:
Yeah, my bad.